### ISSUE TITLE
Which Bullet version should i use?

### ISSUE THREAD

**Issue by foeser** [kbogert/falconunity#2]
Hello,

I downloaded the current source and compiled the solution with Bullet3 and 2.82. 

I get an #QNAN when I tried to ready the position of the godobject (FalconUnity.cs::700). So I then stepped to the native falcon unity source and found out that the problem is somewhere in Bullet. The RigidBody class (btRigigBody) has a method "predictIntegratedTransform(...)" which get called several times in one simulation step. Somehow (and sadly I didn't found out why) the linearVelocity member in this class which is used in the "predictIntegratedTransform(...)" function get set to NAN . This leads to an incorrect origin (btTransform.h::152) set during the transform. And this results in the QNAN for the godobjectPos in falconunity.cpp::267. 

So i tested the Bullet3 and Bullet2.82 version. Which version did you use for v0.2?

Thanks in advice!

**Comment by kbogert** [comment]
I think I finally have an answer for this.  The project was developed using Bullet 2.80 but I ran into this bug (which is apparently documented as issue 665, somewhere ...).

This bug is fixed in Bullet commit f5704d59816b44c7f5ddc01d2cc34668bff80ddc , which is unfortunately after 2.81.  2.82 should have this fix, so I'm not sure why that's not working (some other issue maybe?)

You could use 2.80 and apply the following patch:
```
diff --git a/src/LinearMath/btQuaternion.h b/src/LinearMath/btQuaternion.h
index ee79f6e..8bdb2e5 100644
--- a/src/LinearMath/btQuaternion.h
+++ b/src/LinearMath/btQuaternion.h
@@ -284,7 +284,7 @@ public:
          btAssert(magnitude > btScalar(0));
 
     btScalar product = dot(q) / magnitude;
-    if (btFabs(product) != btScalar(1))
+    if (btFabs(product) < btScalar(1))
                {
       // Take care of long angle case see http://en.wikipedia.org/wiki/Slerp
       const btScalar sign = (product < 0) ? btScalar(-1) : btScalar(1);
```

**Comment by foeser** [comment]
Hey Kenneth,

thanks for your reply. I don't work with the Falcon anymore, but anyway thanks again for you help!
